spec/workers/postcodes_collection_worker_spec.rb

```diff
@@ -0,0 +1,24 @@
+RSpec.describe PostcodesCollectionWorker do
+  describe "#perform" do
+    def perform
+      described_class.new.perform
+    end
+
+    context "where there are postcodes to be refreshed" do
+      before do
+        allow(Postcode).to receive(:select).and_return(%w[E18QS E18QL])
+      end
+
+      it "creates a worker for each postcode" do
+        expect(ProcessPostcodeWorker).to receive(:perform)
+          .with("E18QS")
+        expect(ProcessPostcodeWorker).to receive(:perform)
+          .with("E18QL")
+
+        # expect(sleep).to receive(1)
```

Comment:
Still trying to find a way how to test that the worker is waiting 1 second before each postcode. I tried to check that the `perform` call is not taking less than 2s but it still works if I remove the sleep.